src/components/utils/test/thread_test.cc

```diff
@@ -72,6 +74,76 @@ class TestDelegate : public threads::ThreadDelegate {
   bool check_value_;
 };
 
+class InfiniteMainTestDelegate : public threads::ThreadDelegate {
+ public:
+  InfiniteMainTestDelegate()
+      : notified_(false), notify_running_(false), check_value_(false) {}
+
+  void threadMain() {
+    {
+      AutoLock test_lock(test_mutex_);
+      check_value_ = true;
+      notify_running_ = true;
+      cond_var_.Broadcast();
+    }
+
+    while (!notified_) {
+      threads::sleep(1);
```

Comment:
@AGaliuzov @AGritsevich Adding the second cond var will require also adding another bool variable for the Wait otherwise the notification wakes up sometimes the wrong cond var. I think this way is better than adding a second cond var.